### ISSUE TITLE
Boarding doors don't open at walk-in gates

### ISSUE THREAD

**Issue by robbspeed** [Fragtality/Fenix2GSX#92]
'Improved Door-Handling for Stairs-only to only open when all Stairs stop operating'

Not an issue if the pax deplane to a bus.  But the doors don't open if they will walk to the terminal.

I get the deboarding requested, stairs attach, and then deboarding started.  But the doors don't open unless I open them from the EFB.

Not sure what else to add.  First time posting here.

Thanks

**Comment by Fragtality** [comment]
> Not sure what else to add. First time posting here.

Logs - as the Support Policy in the Readme says.
And in Addition for that Situation, a Screenshot of the GUI to check the reported State Variables.

For Integrations there is no such Thing like a Walk-In Gate. The Spot has Jetways (FSDT_GSX_JETWAY not equal 2) and/or has Stairs (FSDT_GSX_STAIRS not equal 2) and Fenix2GSX reacts to them. And that is pretty much it.

The Change Line you are referring to only included FSDT_GSX_OPERATESTAIRS_STATE in the Logic, because for some Reasons after an GSX Update (that had nothing to do with Stairs), it started to report the Stairs as connected (==5) although they where still moving. 
The Situation you described can only happen if GSX had a brown out again, or something somewhere else had gone wrong and Fenix2GSX wasn't in the Arrival State. Else it would have just open the Doors (or at least the Rear Door if for some uncomprehendable Reason GSX would have reported a Jetway present)

**Comment by robbspeed** [comment]
Thanks very much for the reply.  For the most part, I'm understanding your explanation.  

Again, it only happens when the passengers are to walk to the terminal.  When they wait for a bus, whatever glitch I have sorts itself out in the extra few seconds it takes for the bus to arrive.

I'll attach logs and the GUI screenshot from my one of my upcoming flights in the next 24-36 hours.  Thank you again.

**Comment by Fragtality** [comment]
Aircraft, Scenery, Gate, Profile?

You know the most effective Way to get anything solved, is when a Developer can actually reproduce the Issue and experience it for his/her self ;)

**Comment by robbspeed** [comment]
I almost wanted to say everywhere, but I'll step that back,

The airports I remember are Aerosoft EDDB, DD EPKK, FlyTampa EHAM/EKCH,  Bmw LICC, Orbx LYTV.  I haven't tried at a standard default airport.

It's the only 'break' in the flow for me.  Once I open the doors via the EFB, all of my options resume.  Fueling, Boarding, Pushback... etc for the next flight.

I will have another flight completed by latest 2055z and report back soon after.  Thanks again.

**Comment by Fragtality** [comment]
> The airports I remember are Aerosoft EDDB, DD EPKK, FlyTampa EHAM/EKCH, Bmw LICC, Orbx LYTV. I haven't tried at a standard default airport.

I have most of them - all except LICC and LYTV. So Stand and Profile?
Please don't bother me with default Airports (and I'd recommend the same to you) - they are just 💩 

<br/>

> It's the only 'break' in the flow for me. Once I open the doors via the EFB, all of my options resume. Fueling, Boarding, Pushback... etc for the next flight.

Sure, there it could be worse. But if it is an Issue, it is an Issue (well ... to be honest we're still figuring out if it is an Issue I need to care about or just an Issue of GSX-Jenga-Code-Collapsing or an User-Error 😅 )

**Comment by robbspeed** [comment]
I use profiles from various sources, flightsim.to, rwprofiles, or even the original that may come with the scenery.  It would be hard to remember a particular stand. I can only say it's happened with every Walk-in/No bus stand I've used.

About to load up for LSGG-EDDB.  And as much effort I've put into this thread, I'm sure the doors will open this time.  😅
But I'll report back in a few hours with the logs and screens.

Thanks again so much for your time. I really appreciate it!

**Comment by robbspeed** [comment]
[https://cdn.discordapp.com/attachments/770345154003075084/1287548557510967296/fenix001.png?ex=66f1f289&is=66f0a109&hm=3f0fd3b86d5cbae13d619fc2559974ed8d4fada9d80b3d5fd567bd2d10df8dcc&](url)
Fenix2GSX Intial start-up at EDDB.

[https://cdn.discordapp.com/attachments/770345154003075084/1287548611747381288/fenix002.png?ex=66f1f296&is=66f0a116&hm=a05bb9a5512a13bd17dd8b0f99df5fb5dc90b40dca521c77b88fc2f3492f1254&](url)
Deboarding in EKCH, you can see the passengers but the doors didn't open.

[https://cdn.discordapp.com/attachments/770345154003075084/1287548671876927588/fenix003.png?ex=66f1f2a4&is=66f0a124&hm=789080717c4848ba9d82ddf5274f4ecdebc6ff1d0c434180ca356eaa0f513256&](url)
Deboarding fully completed with doors never opening.

[Fenix2GSX20240922.log](https://github.com/user-attachments/files/17091551/Fenix2GSX20240922.log)

Thanks again for the time.

**Comment by robbspeed** [comment]
testing a link
https://imgur.com/0vFZRDq

**Comment by Fragtality** [comment]
And you surely have the Auto-Door and Auto-Jetway Simulation turned off in the EFB, as required in the Readme?

Else: really a tough one, can't quite figure out where it could go wrong 🤔
There is nothing abnormal in the Logs, the State changed as expected.
Sure, there are some Automations disabled (like Automatic Jetway/Stair Operation), but the Door Handling Code isn't tied to that.
The Jetway and Stair State is also correctly reported, so it could be something with the other Variables used in the Check that could prevent Fenix2GSX from opening the Doors - but there again, I really haven't an Idea why they could have the wrong Values at that Point in Time.

Do you ever use Stands with a Jetway? Maybe it it is something when both Departure and Arrival are handled with Stairs only? (I would have no Idea why that should be a Problem, but at least it would be a clue ^^)

I mean maybe it is even one of this GSX Things? Does this also happen when you manually restart Couatl after Touch-Down?

**Comment by Fragtality** [comment]
Oh btw: you can also directly attach (or even paste directly) Images here on GitHub ;)

**Comment by robbspeed** [comment]
> And you surely have the Auto-Door and Auto-Jetway Simulation turned off in the EFB, as required in the Readme?
I would surely love to say yes.  But I'll load up the sim in about 1 hour and confirm.

> Do you ever use Stands with a Jetway? Maybe it it is something when both Departure and Arrival are handled with Stairs only?
I've tried a lot of different scenarios.  Its only ever on arrival and having to walk to the terminal where the doors don't open.

The more I type, I have this feeling about the EFB settings not being right.  I know I have turned them off before, but Fenix has been updated a few times recently.  Even today there's an update to run before my next flight.  Maybe the settings in EFB got touched without my doing so.  I'll report back in the next hour

And thanks for the GitHub tips  :-)

**Comment by Fragtality** [comment]
Update: could replicate it on EKCH F8 directly with the testArrival Setting and it is an GSX Issue.

The operating Stairs State (Variable) doesn't ever change, it is constantly at 1. And exactly that Variable is needed to change for the Door Handling Code.
So you indeed guessed right, it is because of the Change you mentioned.

So now the Question is, how to handle that:
- Convincing Umberto that GSX might do something wrong here is a difficult Task.
- Just reverting the Change to not listen for the operating Stairs leads again to the Situation of open Doors in free Air - and I really really hate that
- Don't listening to a Change in the operating Stairs would also yield the same Result AND would take the Ability for some Users to manually handle their Doors (since F2G would then directly open or close it again)

**Comment by robbspeed** [comment]
Ah got it! Whew, I'm not crazy then, lol

Ok, well good enough to know that in those parking situations I'll just remember to open the doors on my own.

Thanks so much for your time!

**Comment by robbspeed** [comment]
> * Just reverting the Change to not listen for the operating Stairs leads again to the Situation of open Doors in free Air - and I really really hate that

Yeah so do I.
I'm happy to open the doors manually in the meanwhile.

**Comment by Fragtality** [comment]
Then let's wait where Option 1 might take us, just reported in the Discord 😅

**Comment by Fragtality** [comment]
Well, apparently Umberto decided to ignore yet another Issue although it was correctly described and all Information was given (from the Source he trusts).

So I decided to try to workaround GSX' Behavior (like so often) - how does that Build work for you?
https://www.mediafire.com/file/exyeiojx1rrfw7g/Fenix2GSX-Installer-v0.4.14.exe/file

**Comment by robbspeed** [comment]
Hey I'm sorry for the late reply.  I'm not sure I'm have time to test it today.  When I do, is there shortcut without having to do a complete flight?

**Comment by Fragtality** [comment]
There's no Rush 😉 

Yes, there are Ways to do a quick-test - that's how I could replicate it and also tested the Workaround.
But it is not 100% the same, so it's always a good Idea to test under real Flight Conditions.

If you test, please use the Build I just uploaded: https://www.mediafire.com/file/n8ak4ka9iwm21ju/Fenix2GSX-Installer-v0.4.14.exe/file
I have added another Workaround for a Fenix Bug.

**Comment by robbspeed** [comment]
Ok got it.  I had a rough night of drinks lol... but I'm gonna have some coffee and get started on this now.  Just installed 4.14 so I'll report back in a couple hours.  Thanks again!

**Comment by robbspeed** [comment]
Well, it worked this time!  Same flight EDDB-EKCH.  Thanks so much for looking into this for me.  Log attached in case there's anything you wanted to see.

[Fenix2GSX20240925.log](https://github.com/user-attachments/files/17138358/Fenix2GSX20240925.log)


All the best,
Rob

**Comment by Fragtality** [comment]
Thanks for testing ;)

Log looks normal except one Line:
`Plane not loaded as expected after Boarding, correcting (PaxCurrent 138 | CargoCurrent 248.17) `
So somehow GSX managed to report Boarding as finished before reporting loaded Cargo to 100% - but luckily there is already a "Fail-Safe" for that built-in 😅

**Comment by robbspeed** [comment]
It is very interesting to view the logs and see how much is going on in the background. 

> Plane not loaded as expected after Boarding, correcting (PaxCurrent 138 | CargoCurrent 248.17)
I'm about to start up for 2 flights now and see if this is consistently happening.

Not to change the topic.  One thing I noticed is on a new start of F2G, the fuel is set to 3000kg even as I have the option checked to remember fuel from the last flight.  This isn't a huge thing for me, just thought to let you know.

Ok time to go flying :-)

**Comment by robbspeed** [comment]
Thank you again!  :)

**Comment by Fragtality** [comment]
Finally got an Answer - what is your "Automated staircases" Setting (GSX Settings)?

**Comment by robbspeed** [comment]
![gsx](https://github.com/user-attachments/assets/1adb30b8-d7f4-4641-b8c0-1f9eeea806c2)

I have them partially enabled currently.  I can't remember why I did that, but I know for certain it was something related to a much earlier version of F2G.

Is there something else I should try?  I dock the stairs myself, before starting F2G, and I like to keep attached until I turn the beacon for pushback/start up.

![f2g](https://github.com/user-attachments/assets/aff5cc05-a390-4499-920a-0765523fbbb4)

**Comment by Fragtality** [comment]
> I have them partially enabled currently. I can't remember why I did that, but I know for certain it was something related to a much earlier version of F2G.

Yeah, that Setting is still the Recommendation for Fenix2GSX since that yields the best "Result" / "Experience". But as the Readme says, it should work with all Settings.

<br/>

> Is there something else I should try? I dock the stairs myself, before starting F2G, (initial start of MSFS) and I like to keep attached until I turn the beacon for pushback/start up.

Not at the Moment - the current Goal is to get out why Umberto can't reproduce it although it is something that consistently happens for us. He said that it might be related to that Setting.

<br/>

One Question out of curiosity: What is the Reason to disable Reposition in Fenix2GSX but enable it in GSX?

**Comment by robbspeed** [comment]
I hadn't looked at my GSX settings in a long while, until you asked me for the Stairs option.  Reposition in GSX was prob a new addition in one of his updates.  I'll turn off it next time I fly.  Some things I just like to do manually before I start the automation.  Sorry for the late answer.  I'll prob be offline the next 24 hours. But Tuesday and Wednesday I'll be around.  All the best!

**Comment by Fragtality** [comment]
> I hadn't looked at my GSX settings in a long while, until you asked me for the Stairs option. Reposition in GSX was prob a new addition in one of his updates. I'll turn off it next time I fly. Some things I just like to do manually before I start the automation. Sorry for the late answer. I'll prob be offline the next 24 hours. But Tuesday and Wednesday I'll be around. All the best!

Ah gotcha. Yeah makes Sense, an Reposition might not always be wanted 👍 

FYI: The new GSX Feature needed some Changes, so here is the current Dev-Build:
https://www.mediafire.com/file/onivvm86edi8jg4/Fenix2GSX-Installer-v0.4.15.exe/file
(Just so that you have it whenever you want to fly the Fenix again 😉 )

**Comment by robbspeed** [comment]
Hey thanks very much for providing the updates.  I just installed it but I prob won't start flying until early tomorrow.

**Comment by robbspeed** [comment]
Hahaha thanks for taking the time to really explain it fully  :-)



> I apologize for being so dumb to assume that a Variable named "FSDT_GSX_OPERATESTAIRS_STATE" would have anything to do with the Operating State of the Stairs!

This part really hit!  😂

**Comment by Fragtality** [comment]
No Problem ... I mean partly the Intention was to get some "Sanity Check" for me - so it seems I'm not completely crazy yet. Good News! 😂

**Comment by Fragtality** [comment]
Closed => Update now pushed as Release :)
